### PR TITLE
nixos/acme: more 20.03 release notes updates

### PR DESCRIPTION
**Pull request by rkoe**
###### Motivation for this change
Failures during 20.03 upgrade.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/doc/manual/release-notes/rl-2003.xml [hunk 1]
```diff
@@ -1145,11 +1145,29 @@ systemd.services.nginx.serviceConfig.User = lib.mkForce "root";
...
+    <para>Please note:
+     <itemizedlist>
+      <listitem><para>Certificates will be regenerated on activation, no account or certificate will be migrated from simp-le.
+        In particular private keys will not be preserved. However, the credentials for simp-le are preserved and
+        thus it is possible to roll back to previous versions without breaking certificate generation.
+        Note also that in contrary to simp-le a new private key is recreated at each renewal by default, which can
+        have consequences if you embed your public key in apps.
+       </para>
+       <para>
+        A separate Let's-Encyrpt-account is registered for each distinctive
```
Comment (flokli) [inline nixos/doc/manual/release-notes/rl-2003.xml:1158]:
```suggestion
        A separate Let's-Encrypt-account is registered for each distinctive
```

Comment (flokli) [reply nixos/doc/manual/release-notes/rl-2003.xml:1158]: Also, the rest of the changelog/documentation seems to be pretty `Let's-Encrypt` unspecific - the module uses the ACME protocol, and Let's-Encrypt is the default provider. I'd propose to reword this to match this style.

File: nixos/doc/manual/release-notes/rl-2003.xml [hunk 2]
```diff
@@ -1145,11 +1145,29 @@ systemd.services.nginx.serviceConfig.User = lib.mkForce "root";
...
+        <literal>security.acme.email</literal>/<literal>security.acme.certs.&lt;name&gt;.email</literal>.
+        If you have several certificates, this may trigger Let's-Encrypt account rate limits during the update
```
Comment (flokli) [inline nixos/doc/manual/release-notes/rl-2003.xml:1160]: ACME here, too.

File: nixos/doc/manual/release-notes/rl-2003.xml [hunk 3]
```diff
@@ -1145,11 +1145,29 @@ systemd.services.nginx.serviceConfig.User = lib.mkForce "root";
...
+        (currently max. 10 accounts each 3h). To prevent this, you can use the same email-address for all your
```
Comment (flokli) [inline nixos/doc/manual/release-notes/rl-2003.xml:1161]: email address

File: nixos/doc/manual/release-notes/rl-2003.xml [hunk 4]
```diff
@@ -1145,11 +1145,29 @@ systemd.services.nginx.serviceConfig.User = lib.mkForce "root";
...
+        certificates.
+       </para>
+      </listitem>
+      <listitem><para>By default, now "ec384"-certificates are created instead of rsa-certificates.
```
Comment (flokli) [inline nixos/doc/manual/release-notes/rl-2003.xml:1165]:
```suggestion
      <listitem><para>By default, now "ec384"-certificates are created instead of RSA certificates.
```

File: nixos/doc/manual/release-notes/rl-2003.xml [hunk 5]
```diff
@@ -1145,11 +1145,29 @@ systemd.services.nginx.serviceConfig.User = lib.mkForce "root";
...
+        You may want to change this by setting <literal>security.acme.certs.&lt;name&gt;.keyType</literal>
```
Comment (flokli) [inline nixos/doc/manual/release-notes/rl-2003.xml:1166]: This should be a link referring to the option specified here.

File: nixos/doc/manual/release-notes/rl-2003.xml [hunk 6]
```diff
@@ -1145,11 +1145,29 @@ systemd.services.nginx.serviceConfig.User = lib.mkForce "root";
...
+        back to e.g. <literal>rsa4096</literal>, especially if you are using the certificate for a mailserver.
```
Comment (flokli) [inline nixos/doc/manual/release-notes/rl-2003.xml:1167]: Can you elaborate a bit on this? Why do my mailservers need different certificates?

**Review by flokli — CHANGES_REQUESTED** [review]
Some notes. Also, please rebase the commit on top of master, squash it to one commit, and update the PR title to comething matching `CONTRIBUTING.md`.

**Comment by flokli** [comment]
cc @NixOS/acme for the content.

File: nixos/doc/manual/release-notes/rl-2003.xml (hunk 6 above)
Comment (rkoe) [reply nixos/doc/manual/release-notes/rl-2003.xml:1167]: For interoperatbility, mailservers need RSA certificates.
There are several mailservers out there, which do not yet support ec384-certificates. Now, if you use ec384-certificates, these mailservers cannot send mail to you and send bounce-messages back to the sender (even if encryption is optional in your mailserver). So, if you do this, you will lose mails, cause bounces and be unsubscribed from mailinglists.

So, for a mailserver, you *must* have an RSA certificate.
See also: http://postfix.1071664.n5.nabble.com/TLS-problem-no-shared-cipher-tp105970p105979.html

Comment (flokli) [reply nixos/doc/manual/release-notes/rl-2003.xml:1167]: Thanks to the explanation! We should add a small part of it to the `security.acme.certs.<name>.keyType` option, WDYT?

**Comment by arianvp** [comment]
Content is fine LGTM!

File: nixos/doc/manual/release-notes/rl-2003.xml (hunk 6 above)
Comment (emilazy) [reply nixos/doc/manual/release-notes/rl-2003.xml:1167]: Note that on master (but not 20.03) the default key type is `ec256` instead; see https://github.com/NixOS/nixpkgs/pull/83121#issuecomment-630260135.

Comment (flokli) [reply nixos/doc/manual/release-notes/rl-2003.xml:1167]: We backported the change to use `ec256` to 20.03 as well, as that's what `lego` will default to in their next version: https://github.com/go-acme/lego/issues/1091

**Comment by flokli** [comment]
@rkoe can you address the proposed changes?
